src/plugins/liveobjects/livemap.ts

```diff
@@ -77,13 +77,16 @@ export class LiveMap<T extends API.LiveMapType> extends LiveObject<LiveMapData,
 
   /**
    * Returns the value associated with the specified key in the underlying Map object.
-   * If no element is associated with the specified key, undefined is returned.
-   * If the value that is associated to the provided key is an objectId string of another Live Object,
-   * then you will get a reference to that Live Object if it exists in the local pool, or undefined otherwise.
-   * If the value is not an objectId, then you will get that value.
+   *
+   * - If no entry is associated with the specified key, `undefined` is returned.
+   * - If map entry is tombstoned (deleted), `undefined` is returned.
+   * - If this map object is tombstoned, `undefined` is returned.
```

Comment:
It doesn't look like this method handled this case?

Reply:
```typescript
    if (refObject.isTombstoned()) {
      // tombstoned objects must not be surfaced to the end users
      return undefined as T[TKey];
    }
```
from the [lines below](https://github.com/ably/ably-js/pull/1934/files#diff-2c2d2b2e45367c0ba8c03880c978c1dc73ac8065b20b0887e36e6a56103740e7R114-R117).
probably just missed it among other changes?

Reply:
oh, sorry

Reply:
if "this map object"

Reply:
when we tombstone an object, we remove all its data. so a tombstoned map doesn't have anything in its `_dataRef`, this way any element won't be present in the map.
so `If this map object is tombstoned, `undefined` is returned.` is an implicit consequence of tombstoning a map

Reply:
I can add a comment to the method body to clarify this behavior

Reply:
Yes please, I think that's worth calling out. Thanks